torch/_decomp/decompositions.py

```diff
@@ -987,6 +987,8 @@ def _softmax(x: Tensor, dim: int, half_to_float: bool):
     computation_dtype, result_dtype = utils.elementwise_dtypes(
         x, type_promotion_kind=utils.ELEMENTWISE_TYPE_PROMOTION_KIND.DEFAULT
     )
+    if x.numel() == 0:
+        return torch.zeros_like(x, dtype=result_dtype)
```

Comment:
See that the cast to `result_type` just happens when `not half_to_float`.

To avoid these issues, I would suggest to simply set `x_max = zeros_like(x)` in this case, rather than trying to completely short cut the computation in this case.

Reply:
thanks! let me update that.